### DISCUSSION TITLE
How to pass useTextField props to parent element instead of input element in React form using React Aria?

### DISCUSSION THREAD

**Discussion by yaman3bd** [adobe/react-spectrum#4522] [Q&A] [answered]
I'm trying to use the`useTextField` hook in my React form,
but I'm having trouble passing the errorMessage, description, and label props to the parent element instead of directly to the Input element.
I want to pass my props to the parent element because it would allow me to maintain consistent styling across all form elements.
If I handle error messages, descriptions, labels, etc... in each form element individually, I would need to repeat the same code for every form element, which is not ideal.

here is how I would pass the props to the parent element:
```tsx
import React from "react";

const App = () => {
  return (
    <Form.Group
      label="some-label"
      errorMessage="message"
      description="help"
      className="mb-0"
    >
      <Form.Input
        placeholder="placeholder"
        onChange={(e) => {}}
        required
      />
    </Form.Group>
  );
};

export default App;
```
in this way, I could handle the error message, description, and everything common in one place which is the parent element, and then I can focus on the input, select, checkbox, etc... behavior in separate components instead of having to handle everything in common in each component

and here is how I should pass the props if I used `useTextField` hook:
```tsx
import React from "react";

const App = () => {
  return (
    <Form.Group
      className="mb-0"
    >
      <Form.Input
	label="some-label"
	errorMessage="message"
    	description="help"
        placeholder="placeholder"
        onChange={(e) => {}}
        required
      />
    </Form.Group>
  );
};

export default App;
```
but I found out that I have to handle things like error messages in every single component I have

Any help you can provide would be greatly appreciated. Thank you!

**Comment by snowystinger** [comment, marked as answer]
I don't quite understand how your ideal case works. What happens if you have multiple inputs in the group? How do you distinguish which message/label/etc gets applied to each input? The way each of those hooks (ie useTextfield) works, is it generates an id for the label, an id for the description, an ide for the error message. And then it applies all those ids to the field in their correct aria attributes. It also generates an id for the field, and applies it to the correct attributes for the label. So there is a lot of interconnectivity for it to be accessible.

I think instead of handling it at the group, you should create a common wrapper for each input. We do this in React Spectrum components with the Field component.
Then when you create the Checkbox component, inside, it renders the common Field in addition to the corresponding input type element.

**Reply by yaman3bd** [reply]
in each group, only one input is allowed, I had to handle the common props in the parent element because I am using third-party components like `react-select`, so I have the message/label/etc already handled and I do not have to do anything.
for example, I have only to handle `Select` component specifics:
(I am not handling anything related to accessibility)
```tsx
import React from "react";

import Select from "react-select";

const App = () => {
  return (
    <div className="flex flex-col">
      <Form.Group
        label="some-label"
        errorMessage="message"
        description="help"
        className="mb-0"
      >
        <Form.Input
          placeholder="placeholder"
          onChange={(e) => {}}
          required
        />
      </Form.Group>
      <Form.Group
        label="some-label"
        errorMessage="message"
        description="help"
        className="mb-0"
      >
        <Select
          defaultValue={["value", "value"]}
          name="colors"
          options={[
            "value",
            "value",
            "value",
          ]}
          className="basic-multi-select"
        />
      </Form.Group>
    </div>
  );
};

export default App;
```
and you are right if I am going to add accessibility, it doesn't make sense, I tried in many ways to find a convenient solution but could find one,
I checked the source code of the Field component, it is using Context Provider to share the props to the child elements, and there was a custom solution using `slots`.
I will try to implement it in my components. but for now, I will just do with the default setup and pass everything to the component, but make the common components reusable.
thanks a lot for your time answering me, and giving me a better view of the issue I had.
